imagepruner/v1/types.go

```diff
@@ -0,0 +1,97 @@
+// successful and failed finished jobs to retain.
+type PrunerHistory struct {
+	// SuccessfulJobsHistoryLimit specifies how many successful finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit" protobuf:"bytes,1,opt,name=successfulJobsHistoryLimit"`
+	// FailedJobsHistoryLimit specifies how many failed finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit" protobuf:"bytes,2,opt,name=failedJobsHistoryLimit"`
+}
+
+type ImagePrunerSpec struct {
+	// Schedule specifies when to execute the job
+	// Uses standard cronjob syntax: https://wikipedia.org/wiki/Cron
+	// Defaults to 0 0 * * *
+	// +optional
+	Schedule string `json:"schedule" protobuf:"bytes,1,opt,name=schedule"`
+	// Suspend specifies wether or not to suspend subsequent executions of this cronjob.
+	// Defaults to false
+	// +required
+	Suspend *bool `json:"suspend" protobuf:"bytes,2,opt,name=suspend"`
+	// KeepTagRevisions specifies how many tag revisions to keep.
+	// Defaults to 5
+	// +optional
+	KeepTagRevisions *int `json:"keepTagRevisions" protobuf:"bytes,3,opt,name=keepTagRevisions"`
+	// KeepYoungerThan specifies how old an image needs to be for it to be pruned.
+	// Defaults to 96h (96 hours)
+	// +optional
+	KeepYoungerThan string `json:"keepYoungerThan" protobuf:"bytes,4,opt,name=keepYoungerThan"`
+	// Resources defines the resource requests+limits for the registry pod.
+	// +optional
+	Resources *corev1.ResourceRequirements `json:"resources" protobuf:"bytes,5,opt,name=resources"`
+	// Affinity is a group of node affinity scheduling rules.
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity" protobuf:"bytes,6,opt,name=affinity"`
+	// NodeSelector defines the node selection constraints for the registry
+	// pod.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty" protobuf:"bytes,7,rep,name=nodeSelector"`
+	// Tolerations defines the tolerations for the registry pod.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty" protobuf:"bytes,8,rep,name=tolerations"`
+	// StartingDeadlineSeconds specifies how long of a grace period in seconds
+	// to give the job to start if it misses it's scheduled time.
+	// Defaults to 0
+	// +optional
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty" protobuf:"bytes,9,opt,name=startingDeadlineSeconds"`
+	// History specifies how many successful and failed finished jobs to retain.
+	// +optional
+	History PrunerHistory `json:"history,omitempty" protobuf:"bytes,10,opt,name=history"`
+}
+
+// ImagePrunerStatus reports image pruner operational status.
+type ImagePrunerStatus struct {
+	// ObservedGeneration is the last generation change you've dealt with.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration" protobuf:"bytes,1,opt,name=observedGeneration"`
```

Comment:
Add `omitempty` to align with `OperatorStatus` [1]:
```suggestion
	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"bytes,1,opt,name=observedGeneration"`
```


[1] https://github.com/openshift/api/blob/master/operator/v1/types.go#L100

Reply:
updated in next push